src/bindings.ts

```diff
@@ -1,5 +1,5 @@
-import 'reflect-metadata';
```

Comment:
`reflect-metadata` is needed on this file

Reply:
NO

Reply:
Yes? If you take it out Typescript throws errors because it cannot figure out where references to `Reflect` come from.

Reply:
We toyed with having `reflect-metadata` as a peerDependency in package.json as it fits more in that category. (see https://docs.npmjs.com/files/package.json#peerdependencies ) the problem is, starting with npm 3.0 they stopped installing peer dependencies. This means each user would need to know they need to install `reflect-metadata` in their own project for it to work. This wasn't called out in the README and could be missed so it was just easier to make it a dependency.

Reply:
YOu cannot have 2 or 3 in your code, so you have to use it only in your target project.

Reply:
You can (because we do) but if want you can move it to `index.ts` as a singular place.
You should also merge into this PR the latest master so you can run unit tests against your changes. You'll need to add unit tests to validate this functionality
(if you move `reflect-metadata` you'll need to update the unit tests to have a bootstrap that includes it or import it on each file.

Reply:
Let's remove it and change it to a peerDependency in package.json and add documentation to README.md noting that reflect-metadata needs to be installed alongside trafficlight.    This will need to be a major version increase.

Reply:
What tests?

Reply:
@brantw  perfect it's exactly what injectify had to do, because multiple initializations issue.

Reply:
I put a bunch of unit tests in over the weekend to make it easier for you to validate functionality. They are in master right now if you want to merge master into the branch, then update it

Reply:
I don't want to mess up your tests, you can just add `import 'reflect-metadata';` to the base class.